libvips/foreign/jpegsave.c

```diff
@@ -230,7 +241,7 @@ vips_foreign_save_jpeg_target_build( VipsObject *object )
 		jpeg->Q, jpeg->profile, jpeg->optimize_coding, 
 		jpeg->interlace, save->strip, jpeg->no_subsample,
```

Comment:
Let's not pass no_subsample. Instead, a little higher up, have something like:

```C
if arg-set("no_subsample") && !arg_set("subsample_mode")
  subsample_mode = no_subsample ? OFF : ON
```

ie. old code using the `no_subsample` arg is able to set the default for the new `subsample_mode` value.

Reply:
Should I add the check to each `_build(...)` function? currently I only check once at `vips__jpeg_write_target(...)`.

Is there a way to perform the check in single point without code duplication?

Reply:
Yes, add a build method to the base class for jpegsave and do the test after chaining up.

Reply:
tiffsave has something a little like this:

https://github.com/libvips/libvips/blob/master/libvips/foreign/tiffsave.c#L153

A `_build` method on the base class and use of `isset` to set defaults.

Reply:
done